api/tests/integration/infrastructure/repositories/campaign-analysis-repository_test.js

```diff
@@ -189,6 +192,52 @@ describe('Integration | Repository | Campaign analysis repository', function() {
           expect(tubeFileRecommendation.averageScore).to.equal(80);
         });
       });
+
+      context('when there are some participants who shared their contribution and try to improve it', function() {
```

Comment:
j'aurai peut être pu mettre ça dans le it du dessus . ça fait un peu doublon ?

Reply:
hmm non je trouve pas. En revanche, je trouve qu'il y a trop de config dans le before each : on a pas besoin d'autant de participants par exemple :
```
beforeEach(function() {
          const beforeCampaignParticipationShareDate = new Date('2019-01-01');
          const shareDate = new Date('2019-01-02');
          const userWithCampaignParticipation = _createUserWithSharedCampaignParticipation('Fred', campaignId, shareDate, true);
          _createUserWithNonSharedCampaignParticipation('Fred', campaignId);
          databaseBuilder.factory.buildKnowledgeElement({ userId: userWithCampaignParticipation.userId, skillId: 'recUrl1', status: 'validated', createdAt: beforeCampaignParticipationShareDate });
          return databaseBuilder.commit();
        });
```